catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DumpCommand.java

```diff
@@ -376,13 +413,25 @@ protected void printStatus(long count) {
             return null;
         }
 
-        List<MetacardTransformer> metacardTransformerList = new ArrayList<MetacardTransformer>();
+        List<MetacardTransformer> metacardTransformerList = new ArrayList<>();
         for (int i = 0; i < refs.length; i++) {
-
             metacardTransformerList.add((MetacardTransformer) bundleContext.getService(refs[i]));
         }
 
         return metacardTransformerList;
     }
 
+    private QueryResponseTransformer getZipCompression() {
+        List<QueryResponseTransformer> queryResponseTransformerList = getAllServices(
```

Comment:
You could add a service property to your zip compression transformer such as an id and then filter for it using `getAllServices(QueryResponseTransformer.class, filter)` to fetch only that service.